app/models/mentor_material.rb

```diff
@@ -3,13 +3,12 @@
 class MentorMaterial < ApplicationRecord
   include OrderHelper
 
-  # TODO: hopefully remove some of those
-  belongs_to :core_induction_programme, optional: true
-  belongs_to :course_year, optional: true
-  belongs_to :course_module, optional: true
   belongs_to :course_lesson
+  has_one :course_module, through: :course_lesson
+  has_one :course_year, through: :course_module
+  delegate :core_induction_programme, to: :course_year
```

Comment:
whilst you've changed this to delegate cip to course_year can we change the get_core_induction_programme method and just return course_year.core_induction programme?